### PR TITLE
feat: add github actions workflow and removes travis

### PR DESCRIPTION
**Pull request by tdeekens**
> This pull request adds a GitHub Action to perform the tests.

### Background & Context

The workflow is similar to what we have ton TravisCI. It removes some node versions and has the same semantics of probe tests and testing against BrowserStack.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        run: yarn install --frozen-lockfile
+      - name: Build
+        run: npm run build
+      - name: Lint
+        run: npm run lint
+      - name: Test
+        env:
+          TEST_BROWSERSTACK: ${{ startsWith(matrix.node-version, '13') }}
+          TEST_PROBE_ONLY: ${{ github.ref != 'refs/heads/master' }}
+          BS_USERNAME: ${{ secrets.BS_USERNAME }}
```
Comment (tdeekens) [inline .github/workflows/build-and-test.yml:48]: @cure53 I don't see the `Settings`-tab under the repository. Can you add these as secrets on GitHub. You can find them here. I suppose we have to generate a new set on BrowserStack for the GitHub Actions. Still one we got "rid of" TravisCI we can remove the old ones.

CleanShot 2020-04-19 at 11 08 53@2x

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          node-version: ${{ matrix.node-version }}
+      - name: Install
+        run: yarn install --frozen-lockfile
+      - name: Build
+        run: npm run build
+      - name: Lint
+        run: npm run lint
+      - name: Test
+        env:
+          TEST_BROWSERSTACK: ${{ startsWith(matrix.node-version, '13') }}
```
Comment (tdeekens) [inline .github/workflows/build-and-test.yml:46]: For this I like GitHub Actions.

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "build:cjs": "cross-env NODE_ENV=development BABEL_ENV=rollup rollup -c -f cjs -o dist/purify.cjs.js",
     "test:jsdom": "cross-env NODE_ENV=test BABEL_ENV=rollup node test/jsdom-node-runner --dot",
     "test:karma": "cross-env NODE_ENV=test BABEL_ENV=rollup karma start test/karma.conf.js --log-level warn ",
-    "test:ci": "cross-env NODE_ENV=test BABEL_ENV=rollup npm run lint && npm run test:jsdom && npm run test:karma -- --log-level error --reporters dots --single-run --shouldTestOnBrowserStack=\"${TEST_BROWSERSTACK}\" --shouldProbeOnly=\"${TRAVIS_PULL_REQUEST}\"",
+    "test:ci": "cross-env NODE_ENV=test BABEL_ENV=rollup npm run test:jsdom && npm run test:karma -- --log-level error --reporters dots --single-run --shouldTestOnBrowserStack=\"${TEST_BROWSERSTACK}\" --shouldProbeOnly=\"${TEST_PROBE}\"",
```
Comment (tdeekens) [inline package.json:16]: I split out the lint. I gives them as two steps on CI.

**Comment by tdeekens** [comment]
Odd. Failing with the same test as on my other PR. What Am I doing wrong. I branched off master. Hm.

CleanShot 2020-04-19 at 11 11 07@2x

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (cure53) [reply .github/workflows/build-and-test.yml:48]: What values to initialize them with?

**Comment by cure53** [comment]
> Odd. Failing with the same test as on my other PR. What Am I doing wrong. I branched off master. Hm.
> 
> CleanShot 2020-04-19 at 11 11 07@2x

This might be a temporary artifact. will look into this once we got the rest working

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (cure53) [reply .github/workflows/build-and-test.yml:48]: Added them, two set to `true` and one to the username accordingly

**Comment by tdeekens** [comment]
It seems CI (GitHub or Travis) independent while there is no code change made. Maybe I‘ll just try again in a couple of days. Is that what you are saying? 

My plan was:

- GitHub Actions over Travis
- Dependency house keeping
- Documentation improvements in readme
- Tinker with CodeSandbox to provide examples

Would that work for you? It would also imply that we should be able to try to proceed with this first. I wanted to check to get the same browser setup as on Travis. From there we only need the secrets to my knowledge. 

Thanks for the swift response 😎🤘🏼

**Comment by tdeekens** [comment]
@cure53 let me know when you added the secrets. 🤗

**Comment by cure53** [comment]
@tdeekens Just to avoid me doing something stupid, you need me to add the BS_ACCESSKEY value as secret, correct?

**Comment by tdeekens** [comment]
Yes. In the GitHub settings under secrets we need both referenced secrets referenced by the GitHub Action yml. They should be added in clear text. They are never exposed from the UI and can not be read (only deleted) once added. It is a bit different to TravisCI where you commit the encrypted value but most CIs I have worked with recently go with the UI based solution. Sorry, long story: just add it in the project‘s settings as a secret. 😎

**Comment by cure53** [comment]
Done :)

**Comment by tdeekens** [comment]
Cool! What do we do about the failing test here? I remain clueless. Any hint or help would be ace 🐼.

**Comment by cure53** [comment]
> Cool! What do we do about the failing test here? I remain clueless. Any hint or help would be ace panda_face.

As far as I can see, nothing security critical, we can fix that later.

**Comment by tdeekens** [comment]
> As far as I can see, nothing security critical, we can fix that later.

Okay but what are you suggesting. I don't want to merge anything with failing CI 😄. 

Let me try to fix it to see if the BrowserStack integration would work.

**Comment by tdeekens** [comment]
Alright, there we do. This does the trick now. We could take the gamble and merge it and say 👋 TravisCI 😄.

**Comment by cure53** [comment]
Mergiiing! :D

**Pull request merged**